score/deployment_test.go

```diff
@@ -0,0 +1,62 @@
+package score
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/zegl/kube-score/config"
+	ks "github.com/zegl/kube-score/domain"
+	"github.com/zegl/kube-score/scorecard"
+)
+
+func TestServiceTargetsDeploymentStrategyRolling(t *testing.T) {
+	t.Parallel()
+	testExpectedScore(t, "service-target-deployment.yaml", "Deployment Strategy", scorecard.GradeAllOK)
+}
+
+func TestServiceNotTargetsDeploymentStrategyNotRelevant(t *testing.T) {
+	t.Parallel()
+	// Expecting score 0 as it didn't cot rated
+	skipped := wasSkipped(t, config.Configuration{
+		AllFiles: []ks.NamedReader{testFile("service-not-target-deployment.yaml")},
+	}, "Deployment Strategy")
+	assert.True(t, skipped)
+}
+
+func TestServiceTargetsDeploymentStrategyNotRolling(t *testing.T) {
+	t.Parallel()
+	testExpectedScore(t, "service-target-deployment-not-rolling.yaml", "Deployment Strategy", scorecard.GradeWarning)
+}
+
+func TestServiceTargetsDeploymentStrategyNotSet(t *testing.T) {
+	t.Parallel()
+	testExpectedScore(t, "service-target-deployment-strategy-not-set.yaml", "Deployment Strategy", scorecard.GradeWarning)
+}
```

Comment:
`RollingUpdate` is the default strategy, I don't think that we should require users to explicitly set the strategy to get rid of the warning.

If no strategy is set, the grade should be GradeAllOK.

Reply:
Good point, updated to GradeAllOk